misc/languages/langtabl.txt

```diff
@@ -1,236 +1,879 @@
-wxLANGUAGE_TONGA                       to     -                -                                   LTR    "Tonga"
-wxLANGUAGE_TSONGA                      ts     -                -                                   LTR    "Tsonga"
-wxLANGUAGE_TURKISH                     tr_TR  LANG_TURKISH     SUBLANG_DEFAULT                     LTR    "Turkish"
-wxLANGUAGE_TURKMEN                     tk     -                -                                   LTR    "Turkmen"
-wxLANGUAGE_TWI                         tw     -                -                                   LTR    "Twi"
-wxLANGUAGE_UIGHUR                      ug     -                -                                   LTR    "Uighur"
-wxLANGUAGE_UKRAINIAN                   uk_UA  LANG_UKRAINIAN   SUBLANG_DEFAULT                     LTR    "Ukrainian"
-wxLANGUAGE_URDU                        ur     LANG_URDU        SUBLANG_DEFAULT                     LTR    "Urdu"
-wxLANGUAGE_URDU_INDIA                  ur_IN  LANG_URDU        SUBLANG_URDU_INDIA                  LTR    "Urdu (India)"
-wxLANGUAGE_URDU_PAKISTAN               ur_PK  LANG_URDU        SUBLANG_URDU_PAKISTAN               LTR    "Urdu (Pakistan)"
-wxLANGUAGE_UZBEK                       uz     LANG_UZBEK       SUBLANG_DEFAULT                     LTR    "Uzbek"
-wxLANGUAGE_UZBEK_CYRILLIC              uz     LANG_UZBEK       SUBLANG_UZBEK_CYRILLIC              LTR    "Uzbek (Cyrillic)"
-wxLANGUAGE_UZBEK_LATIN                 uz     LANG_UZBEK       SUBLANG_UZBEK_LATIN                 LTR    "Uzbek (Latin)"
-wxLANGUAGE_VALENCIAN                   ca_ES@valencia -        -                                   LTR    "Valencian (Southern Catalan)"
-wxLANGUAGE_VIETNAMESE                  vi_VN  LANG_VIETNAMESE  SUBLANG_DEFAULT                     LTR    "Vietnamese"
-wxLANGUAGE_VOLAPUK                     vo     -                -                                   LTR    "Volapuk"
-wxLANGUAGE_WELSH                       cy     -                -                                   LTR    "Welsh"
-wxLANGUAGE_WOLOF                       wo     -                -                                   LTR    "Wolof"
-wxLANGUAGE_XHOSA                       xh     -                -                                   LTR    "Xhosa"
-wxLANGUAGE_YIDDISH                     yi     -                -                                   LTR    "Yiddish"
-wxLANGUAGE_YORUBA                      yo     -                -                                   LTR    "Yoruba"
-wxLANGUAGE_ZHUANG                      za     -                -                                   LTR    "Zhuang"
-wxLANGUAGE_ZULU                        zu     -                -                                   LTR    "Zulu"
+wxLANGUAGE_ABKHAZIAN                                    ab             ab                        -              -    -    LTR "Abkhazian","\320\260\322\247\321\201\321\203\320\260 \320\261\321\213\320\267\321\210\323\231\320\260"
```

Comment:
Are octal escapes from the original data source? If not, I think using UTF-8 for this file could be better.

Reply:
No, the octal escapes are not from the original data. The original data are in UTF-8. However, since the data are used to generate a C++ source code file, I prefer hexadecimal or octal escapes to keep the character set of the source file to pure ASCII. For example, Visual Studio sometimes does weird things, if the source file encoding is not ASCII. Here I used octal escapes, because they are as compact as hex escapes and don't introduce problems when the following character in the string happens to be a valid hex digit.